pkg/math/oracle/median.go

```diff
@@ -48,6 +52,10 @@ func NewMedianAggregator(logger *zap.Logger, cfg mmtypes.MarketMap) (*MedianAggr
 //
 // The index price cache contains the previously calculated median prices.
 func (m *MedianAggregator) AggregatedData() {
+	m.indexCache = m.GetAggregatedData()
+	m.providerCache = m.GetProviderData()
+
+	m.Lock()
```

Comment:
Place a lock so that the config cannot change till we finish execution with the current one.